### PR TITLE
Feature/kaa 360

### PR DESCRIPTION
**Pull request by andreypanasenko**
Review @dyosick @GamovCoder

### CODE REVIEW

File: examples/econais/ec19d/sdk_example_kaa_client/inc/kaa_client.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+ *
+ * Created on: Feb 16, 2015
+ *    Author: Andriy Panasenko <apanasenko@cybervisiontech.com>
+*/
+
+#ifndef KAA_CLIENT_H_
+#define KAA_CLIENT_H_
+
+typedef struct {
+        unsigned long max_update_time;
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/inc/kaa_client.h:30]: Can we use some platform independent type such as uintN_t?

File: examples/econais/ec19d/sdk_example_kaa_client/inc/kaa_client.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+ *
+  * This is EXAMPLE declaration, specific should be written based on auto generated
+ * sources from Kaa UI.
+ *
+ * @param[in]       kaa_context     Pointer to an Kaa client.
+ * @param[in]       configuration   Pointer to root configuration
+ *
+ * @return Error code..
+ */
+//kaa_error_t kaa_client_configuration_update(kaa_client_t *kaa_client, const kaa_root_configuration_t *configuration);
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/inc/kaa_client.h:124]: Do we need this?

File: examples/econais/ec19d/sdk_example_kaa_client/src/application.c [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+       if (kaa_error) {
+           sndc_printf("Error %d initializing Kaa client \n",kaa_error);
+           return;
+       }
+   }
+   //Initialize Kaa profile.
+   sndc_printf("Creating profile\n");
+   sndc_thrd_delay(TRACE_DELAY * SNDC_MILLISECOND);
+   /*
+    * Profile initialization should be here, below is example.
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/src/application.c:195]: Why are those block commented?

File: examples/econais/ec19d/sdk_example_kaa_client/readme.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+----------------------------------------------------------------------------------------------------------------
+sdk_example_kaa_client
+----------------------------------------------------------------------------------------------------------------
+
+This example shows how to setup and use Kaa client, Generated code of Kaa should be in src directory.
```
Comment (ohamov) [inline examples/econais/ec19d/sdk_example_kaa_client/readme.txt:5]: Why does the world 'generated' starts with upper case letter after the comma?

File: examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c [hunk 5]
```diff
@@ -0,0 +1,823 @@
...
+typedef enum {
+    BOOTSRAP_UNDEF,
+    BOOTSRAP_STARTED,
+    BOOTSRAP_FINISHED
+} bootstrap_client_state_t;
+
+struct kaa_client_t {
+        char *thread_name;
+        bool_t  operate;
+        sndc_sem_t start_semophore;
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c:55]: Please, fix the mistype in the variable name - semaphore.

File: examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c [hunk 6]
```diff
@@ -0,0 +1,823 @@
...
+        sndc_sem_t logging_semophore;
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c:56]: The same as above.

File: examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c [hunk 7]
```diff
@@ -0,0 +1,823 @@
...
+        self->start_semophore = NULL;
+    }
+
+    if (self->logging_semophore) {
+        sndc_sem_destroy(self->logging_semophore);
+        self->logging_semophore = NULL;
+    }
+
+    if (self->operations_channel.context) {
+        self->operations_channel.destroy(self->operations_channel.context);
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c:222]: Is it possible that a channel may be destroyed twice as the SDK' channel manager does the same?

File: examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c [hunk 8]
```diff
@@ -0,0 +1,823 @@
...
+
+    return KAA_ERR_NONE;
+}
+
+/**
+ * Example code for configuration update
+ */
+kaa_error_t kaa_client_configuration_update(kaa_client_t *kaa_client, const kaa_root_configuration_t *configuration)
+{
+    if (!kaa_client || !configuration) {
```
Comment (ohamov) [inline examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c:788]: Please, use KAA_RETURN_IF_NIL2 where it is possible to do this, to be consistent with code style

File: examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c [hunk 9]
```diff
@@ -0,0 +1,823 @@
...
+
+    kaa_logging_record_t * kaa_record = kaa_logging_record_create();
+    if (!kaa_record) {
+        KAA_LOG_ERROR(kaa_client->kaa_context->logger,
+                KAA_ERR_NOT_INITIALIZED,
+                "Failed to allocate log record");
+        return KAA_ERR_NOT_INITIALIZED;
+    }
+
+    kaa_record->body = kaa_string_move_create(record, NULL);
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c:761]: According to the usage of the kaa_client_log_record function, the right way is to use the kaa_string_copy_create version.

File: examples/econais/ec19d/sdk_example_kaa_client/src/application.c [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+             //Logging example, it just log button pressing.
+             //kaa_client_log_record(kaa_client, "Button pressed");
+         } else {
+             testDone  = true;
+             sndc_printf("Button pressed. \n");
+             start_client();
+         }
+         break;
+      }
+
```
Comment (ohamov) [inline examples/econais/ec19d/sdk_example_kaa_client/src/application.c:338]: Please reduce redundant empty lines.

File: examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c [hunk 11]
```diff
@@ -0,0 +1,823 @@
...
+ * Example code for configuration update
+ */
+kaa_error_t kaa_client_configuration_update(kaa_client_t *kaa_client, const kaa_root_configuration_t *configuration)
+{
+    if (!kaa_client || !configuration) {
+        return KAA_ERR_BADPARAM;
+    }
+    sndc_printf("New configuration update.... timeout %d\n", configuration->timeout);
+
+    KAA_LOG_DEBUG(kaa_client->kaa_context->logger,
```
Comment (ghost) [inline examples/econais/ec19d/sdk_example_kaa_client/src/kaa_client.c:793]: Duplicate log messages.

**Comment by andreypanasenko** [comment]
All notes fixed

**Pull request merged**